File: ai-ecosystem/agentic-lifestyle/src/sync_agent/domain/schemas/iam/ai_context_profile.py
"""Iam.Domain.Models.AIContextProfile + BaseAuditableEntity."""

from datetime import datetime
from uuid import UUID

from sync_agent.domain.schemas.common import AuditableEntityFields, DecimalField


class AIContextProfile(AuditableEntityFields):
    """Maps C# AIContextProfile (excludes navigation property User)."""

    user_id: UUID
    adherence_score: DecimalField
    burnout_risk_score: DecimalField
    churn_risk_score: DecimalField
    motivation_score: DecimalField
    recovery_score: DecimalField
    stress_score: DecimalField
    sleep_quality_score: DecimalField
    nutrition_compliance_score: DecimalField
    workout_compliance_score: DecimalField
    peak_energy_time_window: str | None = None
    preferred_intervention_style: str | None = None
    last_burnout_detected_at: datetime | None = None
    last_workout_skipped_at: datetime | None = None
    last_cheat_meal_at: datetime | None = None
    current_mood: str | None = None
    ai_confidence_score: DecimalField
    last_replan_at: datetime | None = None

File: core/SyncPlatform/src/Services/Iam/Iam.Domain/Models/AIContextProfile.cs
using Libs.Shared.Common;

namespace Iam.Domain.Models;

/// <summary>
/// Bối cảnh hành vi cho agent — điểm số rủi ro / tuân thủ; cập nhật từ pipeline EDA hoặc batch ML.
/// </summary>
public class AIContextProfile : BaseAuditableEntity
{
    public Guid UserId { get; set; }

    public virtual User User { get; set; } = null!;

    public decimal AdherenceScore { get; set; }

    public decimal BurnoutRiskScore { get; set; }

    public decimal ChurnRiskScore { get; set; }

    public decimal MotivationScore { get; set; }

    public decimal RecoveryScore { get; set; }

    public decimal NutritionComplianceScore { get; set; }

    public decimal WorkoutComplianceScore { get; set; }

    public string? PeakEnergyTimeWindow { get; set; }

    public string? PreferredInterventionStyle { get; set; }

    public DateTimeOffset? LastBurnoutDetectedAt { get; set; }

    public DateTimeOffset? LastWorkoutSkippedAt { get; set; }

    public DateTimeOffset? LastCheatMealAt { get; set; }

    public string? CurrentMood { get; set; }

    public decimal AIConfidenceScore { get; set; }

    public DateTimeOffset? LastReplanAt { get; set; }
}
